File: testcases/cpp/reverse_string.cpp
#include <iostream>
#include <string>

std::string reverse_string(std::string s) {
    int len = s.length();
    for (int i = 0; i < len / 2; i++) {
        char temp = s[i];
        s[i] = s[len - i - 1];
        s[len - i - 1] = temp;
    }
    return s;
}

int main() {
    std::cout << reverse_string("hello") << std::endl;
    return 0;
}

File: testcases/rust/reverse_string.rs
fn reverse_string(s: &str) -> String {
    let mut chars: Vec<char> = s.chars().collect();
    let len = chars.len();
    let mut i = 0;
    while i < len / 2 {
        let temp = chars[i];
        chars[i] = chars[len - i - 1];
        chars[len - i - 1] = temp;
        i += 1;
    }
    chars.into_iter().collect()
}

fn main() {
    println!("{}", reverse_string("hello"));
}
